app/src/main/java/emu/skyline/settings/SettingsActivity.kt

```diff
@@ -133,18 +133,18 @@ class SettingsActivity : AppCompatActivity() {
                     preferenceFragment.preferenceScreen.forEach { preferenceCategory ->
                         if (creditsCategories.contains(preferenceCategory.key)) {
                             preferenceCategory.isVisible = false
-                        } else {
-                            val queryMatchesCategory = queries.any { preferenceCategory.title?.contains(it, true) ?: false }
-                            // Tracks whether all preferences under this category are hidden
-                            var areAllPrefsHidden = true
-                            (preferenceCategory as PreferenceCategory).forEach { preference ->
-                                preference.isVisible = queryMatchesCategory || queries.any { preference.title?.contains(it, true) ?: false }
-                                if (preference.isVisible && areAllPrefsHidden)
-                                    areAllPrefsHidden = false
-                            }
-                            // Hide PreferenceCategory if none of its preferences match the search and neither the category title
-                            preferenceCategory.isVisible = !areAllPrefsHidden || queryMatchesCategory
+                            return@forEach
```

Comment:
I only suggested the return because I hate keeping track of brackets and indentation. It's more laziness than anything.

Reply:
Honestly, that makes two of us. I had searched if there was a way to continue a for each loop right before posting it.